src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
+                case 1:
+                    this->currentAnimation = NULL;
+                    func_800BDC5C(&this->skelAnime, animations, 0);
+                    break;
+                case 2:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 3;
+                    func_800BDC5C(&this->skelAnime, animations, 3);
+                    break;
+                case 3:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 5;
+                    func_800BDC5C(&this->skelAnime, animations, 5);
+                    break;
+                case 4:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 7;
+                    if ((this->unk218 == 1) || (this->unk218 == 3)) {
+                        func_8019F128(0x3ABA);
+                    }
+                    func_800BDC5C(&this->skelAnime, animations, 7);
+                    break;
+                case 5:
+                    this->currentAnimation = 1;
+                    func_800BDC5C(&this->skelAnime, animations, 1);
+                    break;
+                case 6:
+                    gSaveContext.weekEventReg[0x4B] |= 0x20;
+                    Actor_MarkForDeath(&this->actor);
+                    break;
+            }
+        } else {
+            if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+                switch (this->currentAnimation) {
+                    case 3:
+                        this->currentAnimation = 4;
+                        func_800BDC5C(&this->skelAnime, animations, 4);
+                        break;
+                    case 5:
+                        this->currentAnimation = 6;
+                        func_800BDC5C(&this->skelAnime, animations, 6);
+                        break;
+                }
+            }
+        }
+        switch (this->currentAnimation) {
+            case 3:
+            case 4:
+                func_800B9010(&this->actor, 0x32B7);
+                break;
+            case 5:
+            case 6:
+                func_800B9010(&this->actor, 0x32B9);
+                break;
+            case 7:
+                if ((this->unk218 == 0) || (this->unk218 == 2)) {
+                    func_800B9010(&this->actor, 0x32B9);
+                }
+                break;
+        }
+        func_800EDF24(&this->actor, globalCtx, actionIndex);
+        return;
+    } else if (globalCtx->csCtx.state == 0) {
+        func_80BCF354(this);
+    }
+    this->cutscenes[3] = 0x63;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF8A0.asm")
+void func_80BCFC0C(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF93C.asm")
+    if (this->actor.colChkInfo.health == 1 && !(fabsf(this->actor.yDistToPlayer) >= 80.0f)) {
```

Comment:
Does `(fabsf(this->actor.yDistToPlayer) < 80.0f))` also match?

Reply:
It does not :/